hazelcast-client-new/src/main/java/com/hazelcast/client/impl/ClientServiceFactory.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.impl;
+
+import com.hazelcast.client.config.ClientConfig;
+import com.hazelcast.client.connection.ClientConnectionManager;
+
+public interface ClientServiceFactory {
+
+    ClientConnectionManager createConnectionManager(ClientConfig config, HazelcastClientInstanceImpl client);
```

Comment:
`ClientServiceFactory` is only used to create `ClientConnectionManager`. Do you plan to add other methods here later? Otherwise I suggest you to rename this to `ClientConnectionManagerFactory` or to something similar..